odoo/orm/models.py

```diff
@@ -2996,11 +2998,11 @@ def _has_field_access(self, field: Field, operation: str) -> bool:
         return self.env.user.has_groups(field.groups)
 
     @api.model
-    def _check_field_access(self, field: Field, operation: str) -> None:
+    def _check_field_access(self, field: Field, operation: typing.Literal['read', 'write']) -> None:
         """Check the user access rights on the given field.
 
         :param field: the field to check
-        :param operation: one of ``create``, ``read``, ``write``, ``unlink``
```

Comment:
I really think that create operation should exist for the consistency with the ACL logic. 
Also, it allows people to override `_check_field_access` and have a specific case for the create.

Reply:
"create" could introduce an expectation that you can make it read-only after creation by overwriting check_field_access. While true in technical sense, it can still be modified using sudo.
"unlink" does not make sense, so it will not be in line with record ACLs.

From the perspective of `FIeld`, we have operations *get* (read) and *set* (write). There are no operations relative to creation of current record.

Reply:
> From the perspective of Field, we have operations get (read) and set (write).

Yeah I understand that, but in practice this method is meant to be override. IMO, in that case, we must have a way to distinguish create() and a write().

Reply:
> Yeah I understand that, but in practice this method is meant to be override. IMO, in that case, we must have a way to distinguish create() and a write().

You can actually turn the argument around: an override of `create()` may perform a `write()`, and the intention of that `write()` is to be part of the `create()` ACLs...

Reply:
When inside `create()` you call `write()`, it will check write-ACLs anyways.